### PR TITLE
Add more configuration options to RMT

### PR DESCRIPTION
**Pull request by lask**

### CODE REVIEW

File: lib/rmt.toit [hunk 1]
```diff
@@ -205,34 +204,20 @@ class Channel:
     rmt_config_rx_ pin.num num mem_block_num clk_div flags idle_threshold filter_en filter_ticks_thresh rx_buffer_size
     idle_threshold_ = idle_threshold
     rx_buffer_size_ = rx_buffer_size
-
-  /**
```
Comment (lask) [inline lib/rmt.toit:209]: This was just moved.

File: lib/rmt.toit [hunk 2]
```diff
@@ -296,9 +301,12 @@ The $rx channel must be configured for receiving (see $Channel.config_rx).
 The $tx channel must be configured for transferring (see $Channel.config_tx).
 */
 transfer_and_receive --rx/Channel --tx/Channel signals/Signals max_returned_bytes/int -> Signals:
-  if max_returned_bytes > rx.rx_buffer_size: throw "maximum returned buffer size greater than allocated RX buffer size"
+  if not rx.rx_buffer_size or not rx.rx_clk_div: throw "rx channel not configured"
```
Comment (erikcorry) [inline lib/rmt.toit:304]: This is the same as `not (rx.rx_buffer_size or not rx.rx_clk_div)` which I think is not what you wanted.

File: lib/rmt.toit [hunk 3]
```diff
@@ -296,9 +301,12 @@ The $rx channel must be configured for receiving (see $Channel.config_rx).
...
 
-  result := rmt_transfer_and_read_ tx.num rx.num signals.bytes_ max_returned_bytes
+  if not rx.rx_buffer_size or max_returned_bytes > rx.rx_buffer_size: throw "maximum returned buffer size greater than allocated RX buffer size"
```
Comment (erikcorry) [inline lib/rmt.toit:306]: We are rechecking rx.rx_buffer_size for nullness, which may not be what you want.

File: src/resources/rmt_esp32.cc [hunk 4]
```diff
@@ -153,6 +152,13 @@ PRIMITIVE(config_rx) {
   return process->program()->null_object();
 }
 
+PRIMITIVE(set_idle_threshold) {
+  ARGS(int, channel_num, uint16, threshold)
+  esp_err_t err = rmt_get_rx_idle_thresh(static_cast<rmt_channel_t>(channel_num), &threshold);
```
Comment (erikcorry) [inline src/resources/rmt_esp32.cc:157]: It's called `set`, but it calls `get`.  Is this what you want?

File: lib/rmt.toit (hunk 2 above)
Comment (lask) [reply lib/rmt.toit:304]: Changed it to `not rx.rx_buffer_size and rx.rx_clk_div`.

File: src/resources/rmt_esp32.cc (hunk 4 above)
Comment (lask) [reply src/resources/rmt_esp32.cc:157]: I want `set`, so I can dynamically change the threshold. I should have caught this when I made the second argument a pointer.

File: lib/rmt.toit (hunk 3 above)
Comment (lask) [reply lib/rmt.toit]: Removed the second check.

**Review by erikcorry — APPROVED** [review]

**Pull request merged**